cost/azure/rightsize_sql_instances/azure_rightsize_sql_instances.pt

```diff
@@ -692,15 +713,51 @@ script "js_merged_metrics", type: "javascript" do
 
     if (cpu_metrics != null && instance['recommendationType'] != 'Delete' && param_unused_or_underutilized != "Unused") {
       if (cpu_metrics.length > 0) {
+        minimums = _.pluck(cpu_metrics, 'minimum')
+        maximums = _.pluck(cpu_metrics, 'maximum')
         averages = _.pluck(cpu_metrics, 'average')
+
+        cpu_minimum = null
+        cpu_maximum = 0
         cpu_average = 0
+        cpu_p90 = 0
+        cpu_p95 = 0
+        cpu_p99 = 0
+
+        _.each(minimums, function(item) {
+          if (cpu_minimum == null || item < cpu_minimum) { cpu_minimum = item }
+        })
+
+        _.each(maximums, function(item) {
+          if (item > cpu_maximum) { cpu_maximum = item }
+        })
+
+        if (cpu_minimum == null) { cpu_minimum = 0 }
 
         if (averages.length > 0) {
           cpu_average = _.reduce(averages, function(memo, num) { return memo + num }, 0) / averages.length
         }
 
-        if (cpu_average <= param_stats_underutil_threshold_cpu_value) {
+        if (minimums.length + maximums.length + averages.length > 0) {
+          cpu_p90 = percentile(minimums.concat(maximums, averages), 90)
```

Comment:
```suggestion
          // combine all points from minimums, maximums, averages
          // use percentile() function to get the percentile values from all measured utilization points
          cpu_p90 = percentile(minimums.concat(maximums, averages), 90)
```
Just a comment here to clarify it's already being joined.. my initial thought this was just the `minimums` but then I saw .concat(...)